src/flake8_pytest_style/plugins/unittest_assert.rs

```diff
@@ -0,0 +1,420 @@
+                    Location::default(),
+                    Location::default(),
+                    ExprKind::Call {
+                        func: Box::new(Expr::new(
+                            Location::default(),
+                            Location::default(),
+                            ExprKind::Name {
+                                id: "isinstance".to_string(),
+                                ctx: Load,
+                            },
+                        )),
+                        args: vec![(**obj).clone(), (**cls).clone()],
+                        keywords: vec![],
+                    },
+                );
+                if matches!(self, UnittestAssert::IsInstance) {
+                    Ok(assert(&isinstance, msg))
+                } else {
+                    let expr = Expr::new(
+                        Location::default(),
+                        Location::default(),
+                        ExprKind::UnaryOp {
+                            op: Unaryop::Not,
+                            operand: Box::new(isinstance),
+                        },
+                    );
+                    Ok(assert(&expr, msg))
+                }
+            }
+            UnittestAssert::Regex
+            | UnittestAssert::RegexpMatches
+            | UnittestAssert::NotRegex
+            | UnittestAssert::NotRegexpMatches => {
+                let args = self.arg_hashmap(args, keywords)?;
+                let regex = args.get("regex").ok_or("Missing argument `regex`")?;
+                let text = args.get("text").ok_or("Missing argument `text`")?;
+                let msg = args.get("msg").copied();
+                let re_search = Expr::new(
+                    Location::default(),
+                    Location::default(),
+                    ExprKind::Call {
+                        func: Box::new(Expr::new(
+                            Location::default(),
+                            Location::default(),
+                            ExprKind::Attribute {
+                                value: Box::new(Expr::new(
+                                    Location::default(),
+                                    Location::default(),
+                                    ExprKind::Name {
+                                        id: "re".to_string(),
+                                        ctx: Load,
+                                    },
+                                )),
+                                attr: "search".to_string(),
+                                ctx: Load,
+                            },
+                        )),
+                        args: vec![(**regex).clone(), (**text).clone()],
+                        keywords: vec![],
+                    },
+                );
+                if matches!(self, UnittestAssert::Regex | UnittestAssert::RegexpMatches) {
+                    Ok(assert(&re_search, msg))
+                } else {
+                    let expr = Expr::new(
```

Comment:
You can use `ast::helpers::create_expr` to avoid having to provide the `Location::default()` args, although maybe that's not that helpful of a utility.